bundles/org.eclipse.jface.text/src/org/eclipse/jface/text/source/AnnotationRulerColumn.java

```diff
@@ -517,10 +525,18 @@ private void doubleBufferPaint(GC dest) {
 				fBuffer= null;
 			}
 		}
-		if (fBuffer == null)
-			fBuffer= new Image(fCanvas.getDisplay(), size.x, size.y);
+		if (fBuffer == null) {
+			ImageGcDrawer imageGcDrawer= (gc, imageWidth, imageHeight) -> {
+				doPaint(gc, new Point(imageWidth, imageHeight));
+			};
+			fBuffer= new Image(fCanvas.getDisplay(), imageGcDrawer, size.x, size.y);
```

Comment:
When changing the signature of `doPaint` to accept width/height instead of size, this could be simplified to:
```suggestion
			fBuffer= new Image(fCanvas.getDisplay(), this::doPaint, size.x, size.y);
```

Reply:
Nice improvement, I adapted the affected places